scripts/test_ci.sh

```
@@ -0,0 +1,157 @@
+if [[ -z "$PYTHON" ]]; then
+  warn "Python 3.12+ not found on PATH — falling back to whatever python3 is present"
+  PYTHON="python3"
+fi
+
+# ── Step 1: Syntax check (always runs) ───────────────────────────────────────
+echo ""
+echo "── Syntax check ──────────────────────────────────────────"
+find . -name "*.py" \
+  -not -path "./.venv/*" \
+  -not -path "./.ci-venv/*" \
+  -not -path "./.git/*" \
```

Shellcheck (0.11.0)

[warning] Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames.

(SC2038)